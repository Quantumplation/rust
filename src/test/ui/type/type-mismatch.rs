#![allow(non_camel_case_types)]

trait Qux {}
struct A;
struct B;
impl Qux for A {}
impl Qux for B {}

struct Foo<T, U: Qux = A, V: Qux = B>(T, U, V);

struct foo;
struct bar;

fn want<T>(t: T) {}

fn have_usize(f: usize) {
    want::<foo>(f); //~ ERROR arguments to this function are incorrect
    want::<bar>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<usize>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<usize, B>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<foo>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<foo, B>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<bar>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<bar, B>>(f); //~ ERROR arguments to this function are incorrect
}

fn have_foo(f: foo) {
    want::<usize>(f); //~ ERROR arguments to this function are incorrect
    want::<bar>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<usize>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<usize, B>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<foo>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<foo, B>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<bar>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<bar, B>>(f); //~ ERROR arguments to this function are incorrect
}

fn have_foo_foo(f: Foo<foo>) {
    want::<usize>(f); //~ ERROR arguments to this function are incorrect
    want::<foo>(f); //~ ERROR arguments to this function are incorrect
    want::<bar>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<usize>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<usize, B>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<foo, B>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<bar>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<bar, B>>(f); //~ ERROR arguments to this function are incorrect
    want::<&Foo<foo>>(f); //~ ERROR arguments to this function are incorrect
    want::<&Foo<foo, B>>(f); //~ ERROR arguments to this function are incorrect
}

fn have_foo_foo_b(f: Foo<foo, B>) {
    want::<usize>(f); //~ ERROR arguments to this function are incorrect
    want::<foo>(f); //~ ERROR arguments to this function are incorrect
    want::<bar>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<usize>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<usize, B>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<foo>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<bar>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<bar, B>>(f); //~ ERROR arguments to this function are incorrect
    want::<&Foo<foo>>(f); //~ ERROR arguments to this function are incorrect
    want::<&Foo<foo, B>>(f); //~ ERROR arguments to this function are incorrect
}

fn have_foo_foo_b_a(f: Foo<foo, B, A>) {
    want::<usize>(f); //~ ERROR arguments to this function are incorrect
    want::<foo>(f); //~ ERROR arguments to this function are incorrect
    want::<bar>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<usize>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<usize, B>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<foo>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<foo, B>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<bar>>(f); //~ ERROR arguments to this function are incorrect
    want::<Foo<bar, B>>(f); //~ ERROR arguments to this function are incorrect
    want::<&Foo<foo>>(f); //~ ERROR arguments to this function are incorrect
    want::<&Foo<foo, B>>(f); //~ ERROR arguments to this function are incorrect
}

fn main() {}
