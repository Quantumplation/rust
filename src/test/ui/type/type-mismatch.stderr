error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:17:5
   |
LL |     want::<foo>(f);
   |     ^^^^^^^^^^^^-^
   |                 |
   |                 expected `foo`, found `usize`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({foo});
   |     ^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:18:5
   |
LL |     want::<bar>(f);
   |     ^^^^^^^^^^^^-^
   |                 |
   |                 expected `bar`, found `usize`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({bar});
   |     ^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:19:5
   |
LL |     want::<Foo<usize>>(f);
   |     ^^^^^^^^^^^^^^^^^^^-^
   |                        |
   |                        expected `Foo<usize>`, found `usize`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<usize>});
   |     ^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:20:5
   |
LL |     want::<Foo<usize, B>>(f);
   |     ^^^^^^^^^^^^^^^^^^^^^^-^
   |                           |
   |                           expected `Foo<usize, B>`, found `usize`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<usize, B>});
   |     ^^^^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:21:5
   |
LL |     want::<Foo<foo>>(f);
   |     ^^^^^^^^^^^^^^^^^-^
   |                      |
   |                      expected `Foo<foo>`, found `usize`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<foo>});
   |     ^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:22:5
   |
LL |     want::<Foo<foo, B>>(f);
   |     ^^^^^^^^^^^^^^^^^^^^-^
   |                         |
   |                         expected `Foo<foo, B>`, found `usize`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<foo, B>});
   |     ^^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:23:5
   |
LL |     want::<Foo<bar>>(f);
   |     ^^^^^^^^^^^^^^^^^-^
   |                      |
   |                      expected `Foo<bar>`, found `usize`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<bar>});
   |     ^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:24:5
   |
LL |     want::<Foo<bar, B>>(f);
   |     ^^^^^^^^^^^^^^^^^^^^-^
   |                         |
   |                         expected `Foo<bar, B>`, found `usize`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<bar, B>});
   |     ^^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:28:5
   |
LL |     want::<usize>(f);
   |     ^^^^^^^^^^^^^^-^
   |                   |
   |                   expected `usize`, found `foo`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({usize});
   |     ^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:29:5
   |
LL |     want::<bar>(f);
   |     ^^^^^^^^^^^^-^
   |                 |
   |                 expected `bar`, found `foo`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({bar});
   |     ^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:30:5
   |
LL |     want::<Foo<usize>>(f);
   |     ^^^^^^^^^^^^^^^^^^^-^
   |                        |
   |                        expected `Foo<usize>`, found `foo`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<usize>});
   |     ^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:31:5
   |
LL |     want::<Foo<usize, B>>(f);
   |     ^^^^^^^^^^^^^^^^^^^^^^-^
   |                           |
   |                           expected `Foo<usize, B>`, found `foo`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<usize, B>});
   |     ^^^^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:32:5
   |
LL |     want::<Foo<foo>>(f);
   |     ^^^^^^^^^^^^^^^^^-^
   |                      |
   |                      expected `Foo<foo>`, found `foo`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<foo>});
   |     ^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:33:5
   |
LL |     want::<Foo<foo, B>>(f);
   |     ^^^^^^^^^^^^^^^^^^^^-^
   |                         |
   |                         expected `Foo<foo, B>`, found `foo`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<foo, B>});
   |     ^^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:34:5
   |
LL |     want::<Foo<bar>>(f);
   |     ^^^^^^^^^^^^^^^^^-^
   |                      |
   |                      expected `Foo<bar>`, found `foo`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<bar>});
   |     ^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:35:5
   |
LL |     want::<Foo<bar, B>>(f);
   |     ^^^^^^^^^^^^^^^^^^^^-^
   |                         |
   |                         expected `Foo<bar, B>`, found `foo`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<bar, B>});
   |     ^^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:39:5
   |
LL |     want::<usize>(f);
   |     ^^^^^^^^^^^^^^-^
   |                   |
   |                   expected `usize`, found `Foo<foo>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({usize});
   |     ^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:40:5
   |
LL |     want::<foo>(f);
   |     ^^^^^^^^^^^^-^
   |                 |
   |                 expected `foo`, found `Foo<foo>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({foo});
   |     ^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:41:5
   |
LL |     want::<bar>(f);
   |     ^^^^^^^^^^^^-^
   |                 |
   |                 expected `bar`, found `Foo<foo>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({bar});
   |     ^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:42:5
   |
LL |     want::<Foo<usize>>(f);
   |     ^^^^^^^^^^^^^^^^^^^-^
   |                        |
   |                        expected `Foo<usize>`, found `Foo<foo>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<usize>});
   |     ^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:43:5
   |
LL |     want::<Foo<usize, B>>(f);
   |     ^^^^^^^^^^^^^^^^^^^^^^-^
   |                           |
   |                           expected `Foo<usize, B>`, found `Foo<foo>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<usize, B>});
   |     ^^^^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:44:5
   |
LL |     want::<Foo<foo, B>>(f);
   |     ^^^^^^^^^^^^^^^^^^^^-^
   |                         |
   |                         expected `Foo<foo, B>`, found `Foo<foo>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<foo, B>});
   |     ^^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:45:5
   |
LL |     want::<Foo<bar>>(f);
   |     ^^^^^^^^^^^^^^^^^-^
   |                      |
   |                      expected `Foo<bar>`, found `Foo<foo>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<bar>});
   |     ^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:46:5
   |
LL |     want::<Foo<bar, B>>(f);
   |     ^^^^^^^^^^^^^^^^^^^^-^
   |                         |
   |                         expected `Foo<bar, B>`, found `Foo<foo>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<bar, B>});
   |     ^^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:47:5
   |
LL |     want::<&Foo<foo>>(f);
   |     ^^^^^^^^^^^^^^^^^^-^
   |                       |
   |                       expected `&Foo<foo>`, found `Foo<foo>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({&Foo<foo>});
   |     ^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:48:5
   |
LL |     want::<&Foo<foo, B>>(f);
   |     ^^^^^^^^^^^^^^^^^^^^^-^
   |                          |
   |                          expected `&Foo<foo, B>`, found `Foo<foo>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({&Foo<foo, B>});
   |     ^^^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:52:5
   |
LL |     want::<usize>(f);
   |     ^^^^^^^^^^^^^^-^
   |                   |
   |                   expected `usize`, found `Foo<foo, B>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({usize});
   |     ^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:53:5
   |
LL |     want::<foo>(f);
   |     ^^^^^^^^^^^^-^
   |                 |
   |                 expected `foo`, found `Foo<foo, B>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({foo});
   |     ^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:54:5
   |
LL |     want::<bar>(f);
   |     ^^^^^^^^^^^^-^
   |                 |
   |                 expected `bar`, found `Foo<foo, B>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({bar});
   |     ^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:55:5
   |
LL |     want::<Foo<usize>>(f);
   |     ^^^^^^^^^^^^^^^^^^^-^
   |                        |
   |                        expected `Foo<usize>`, found `Foo<foo, B>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<usize>});
   |     ^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:56:5
   |
LL |     want::<Foo<usize, B>>(f);
   |     ^^^^^^^^^^^^^^^^^^^^^^-^
   |                           |
   |                           expected `Foo<usize, B>`, found `Foo<foo, B>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<usize, B>});
   |     ^^^^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:57:5
   |
LL |     want::<Foo<foo>>(f);
   |     ^^^^^^^^^^^^^^^^^-^
   |                      |
   |                      expected `Foo<foo>`, found `Foo<foo, B>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<foo>});
   |     ^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:58:5
   |
LL |     want::<Foo<bar>>(f);
   |     ^^^^^^^^^^^^^^^^^-^
   |                      |
   |                      expected `Foo<bar>`, found `Foo<foo, B>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<bar>});
   |     ^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:59:5
   |
LL |     want::<Foo<bar, B>>(f);
   |     ^^^^^^^^^^^^^^^^^^^^-^
   |                         |
   |                         expected `Foo<bar, B>`, found `Foo<foo, B>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<bar, B>});
   |     ^^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:60:5
   |
LL |     want::<&Foo<foo>>(f);
   |     ^^^^^^^^^^^^^^^^^^-^
   |                       |
   |                       expected `&Foo<foo>`, found `Foo<foo, B>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({&Foo<foo>});
   |     ^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:61:5
   |
LL |     want::<&Foo<foo, B>>(f);
   |     ^^^^^^^^^^^^^^^^^^^^^-^
   |                          |
   |                          expected `&Foo<foo, B>`, found `Foo<foo, B>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({&Foo<foo, B>});
   |     ^^^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:65:5
   |
LL |     want::<usize>(f);
   |     ^^^^^^^^^^^^^^-^
   |                   |
   |                   expected `usize`, found `Foo<foo, B, A>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({usize});
   |     ^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:66:5
   |
LL |     want::<foo>(f);
   |     ^^^^^^^^^^^^-^
   |                 |
   |                 expected `foo`, found `Foo<foo, B, A>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({foo});
   |     ^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:67:5
   |
LL |     want::<bar>(f);
   |     ^^^^^^^^^^^^-^
   |                 |
   |                 expected `bar`, found `Foo<foo, B, A>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({bar});
   |     ^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:68:5
   |
LL |     want::<Foo<usize>>(f);
   |     ^^^^^^^^^^^^^^^^^^^-^
   |                        |
   |                        expected `Foo<usize>`, found `Foo<foo, B, A>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<usize>});
   |     ^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:69:5
   |
LL |     want::<Foo<usize, B>>(f);
   |     ^^^^^^^^^^^^^^^^^^^^^^-^
   |                           |
   |                           expected `Foo<usize, B>`, found `Foo<foo, B, A>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<usize, B>});
   |     ^^^^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:70:5
   |
LL |     want::<Foo<foo>>(f);
   |     ^^^^^^^^^^^^^^^^^-^
   |                      |
   |                      expected `Foo<foo>`, found `Foo<foo, B, A>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<foo>});
   |     ^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:71:5
   |
LL |     want::<Foo<foo, B>>(f);
   |     ^^^^^^^^^^^^^^^^^^^^-^
   |                         |
   |                         expected `Foo<foo, B>`, found `Foo<foo, B, A>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<foo, B>});
   |     ^^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:72:5
   |
LL |     want::<Foo<bar>>(f);
   |     ^^^^^^^^^^^^^^^^^-^
   |                      |
   |                      expected `Foo<bar>`, found `Foo<foo, B, A>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<bar>});
   |     ^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:73:5
   |
LL |     want::<Foo<bar, B>>(f);
   |     ^^^^^^^^^^^^^^^^^^^^-^
   |                         |
   |                         expected `Foo<bar, B>`, found `Foo<foo, B, A>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({Foo<bar, B>});
   |     ^^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:74:5
   |
LL |     want::<&Foo<foo>>(f);
   |     ^^^^^^^^^^^^^^^^^^-^
   |                       |
   |                       expected `&Foo<foo>`, found `Foo<foo, B, A>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({&Foo<foo>});
   |     ^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/type-mismatch.rs:75:5
   |
LL |     want::<&Foo<foo, B>>(f);
   |     ^^^^^^^^^^^^^^^^^^^^^-^
   |                          |
   |                          expected `&Foo<foo, B>`, found `Foo<foo, B, A>`
   |
note: function defined here
  --> $DIR/type-mismatch.rs:14:4
   |
LL | fn want<T>(t: T) {}
   |    ^^^^    ----
help: provide an argument of the correct type
   |
LL |     want({&Foo<foo, B>});
   |     ^^^^^^^^^^^^^^^^^^^^

error: aborting due to 47 previous errors

For more information about this error, try `rustc --explain E0308`.
